<launch>
     <arg name="verbose_timings"   default="false"/>
     <arg name="publish_tf"        default="true"/>
     <arg name="use_tf_prefix"     default="false"/>
     <arg name="publish_images"    default="true"/>
     <arg name="use_aligned"       default="false"/>
     <arg name="use_float_depth"   default="false"/>
     <arg name="use_8bit_depth"    default="false"/>
     <arg name="post_process"      default="false"/>
     <arg name="get_disparity"     default="false"/>
     <arg name="update_rate"       default="120"/>
     <arg name="cam_fps"           default="30"/>
     <arg name="depth_width"       default="848"/>
     <arg name="depth_height"      default="480"/>
     <arg name="color_width"       default="848"/>
     <arg name="color_height"      default="480"/>

     <node pkg="swanson_sensors" type="camera_d415_node" name="d400_camera_node" respawn="true" output="screen">
          <param name="ns"                   value="$(arg ns)"/>
          <param name="verbose_timings"      value="$(arg verbose_timings)"/>
          <param name="publish_tf"           value="$(arg publish_tf)"/>
          <param name="use_tf_prefix"        value="$(arg use_tf_prefix)"/>
          <param name="publish_images"       value="$(arg publish_images)"/>
          <param name="use_aligned"          value="$(arg use_aligned)"/>
          <param name="use_float_depth"      value="$(arg use_float_depth)"/>
          <param name="use_8bit_depth"       value="$(arg use_8bit_depth)"/>
          <param name="post_process"         value="$(arg post_process)"/>
          <param name="get_disparity"        value="$(arg get_disparity)"/>
          <param name="depth_fps"            value="$(arg cam_fps)"/>
          <param name="depth_width"          value="$(arg depth_width)"/>
          <param name="depth_height"         value="$(arg depth_height)"/>
          <param name="color_fps"            value="$(arg cam_fps)"/>
          <param name="color_width"          value="$(arg color_width)"/>
          <param name="color_height"         value="$(arg color_height)"/>
          <param name="update_rate"          value="$(arg update_rate)"/>
     </node>

     <node pkg="tf" type="static_transform_publisher" name="tf_cam_to_base" args="0.26 0 0 0 0 0 base_link camera_link 100"/>

     <group if="$(arg get_cloud)">
          <!-- Nodelet manager for this pipeline -->
          <node pkg="nodelet" type="nodelet" args="manager" name="depth_proc_manager" output="screen"/>

          <node pkg="nodelet" type="nodelet" name="depth_to_disparity" args="load depth_image_proc/disparity depth_proc_manager --no-bond">
               <remap from="left/image_rect" to="/hytx/camera/depth/image_rect_raw"/>
               <remap from="right/camera_info" to="/hytx/camera/depth/camera_info"/>
               <!-- <remap from="depth_registered/points" to="$(arg cloud_topic)"/>
               <remap from="rgb/image_rect_color" to="$(arg camera_name)/$(arg rgb_image)"/>
               <remap from="rgb/camera_info" to="$(arg camera_name)/$(arg rgb_info)"/> -->
          </node>

     </group>

     <!-- <include if="$(arg use_cam2pc)" file="$(find swanson_sensors)/launch/depth_to_pc.launch">
     <include if="$(arg use_cam2pc)" file="$(find swanson_sensors)/launch/pc_filter.launch">
          <arg name="use_cam2pc"   value="$(arg use_cam2pc)"/>
     </include> -->

</launch>
